Snowflake.Data/Core/Session/SFSession.cs

```diff
@@ -432,8 +439,16 @@ internal RequestQueryContext GetQueryContextRequest()
 
         internal void UpdateDatabaseAndSchema(string databaseName, string schemaName)
         {
-            this.database = databaseName;
-            this.schema = schemaName;
+            // with HTAP session metadata removal database/schema
+            // might be not returened in query result
+            if (databaseName != null)

```

Comment:
I would use String.IsNullOrEmpty even we know empty string it's less likely

Reply:
Done. That was my initial thoughts but then I thought maybe empty string could be a valid value returned from server in the future? Since it's not currently (there is no way to clear the database/schema set in the session without change it to a new one) I think better to make the code more straight forward.